assets/js/theme/global/maintenanceMode.js

```diff
@@ -7,17 +7,43 @@ import $ from 'jquery';
 export default function (maintenanceMode = {}) {
     const header = maintenanceMode.header;
     const notice = maintenanceMode.notice;
+    const password = maintenanceMode.password;
```

Comment:
Good catch @mjschock 

You are injecting the password here https://github.com/bigcommerce/bigcommerce/pull/18694/files#diff-3491c3f0657689614b41beb712cb7681R114

And that is injected in the theme even if the user is not a store admin